From 643f092def8154142fc1f2bce9eb21a7a5ff4756 Mon Sep 17 00:00:00 2001
From: "qipeng.zha" <qipeng.zha@intel.com>
Date: Fri, 16 Jan 2015 00:06:05 +0800
Subject: [PATCH 1/1] Touch: reset touch when probe

Touch will check i2c mode pin when power on
to detect HID or object mode, if BIOS configure
this pin without keep reset pin active, then
touch chip will enter into HID mode even i2c
mode pin is high.
This patch will reset touch when probe to avoid
this such case.

Change-Id: Ic6258185cf43f5e137fd6a03ae792940a0d8446c
Signed-off-by: qipeng.zha <qipeng.zha@intel.com>
---
 drivers/input/touchscreen/atmel_mxt_ts.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/input/touchscreen/atmel_mxt_ts.c b/drivers/input/touchscreen/atmel_mxt_ts.c
index 90f5a9d..c1424ef 100644
--- a/drivers/input/touchscreen/atmel_mxt_ts.c
+++ b/drivers/input/touchscreen/atmel_mxt_ts.c
@@ -3768,6 +3768,12 @@ static int mxt_probe(struct i2c_client *client,
 		gpio = devm_gpiod_get_index(&client->dev, "atml_gpio_rst", 0);
 		if (!IS_ERR(gpio)) {
 			pdata->gpio_reset = desc_to_gpio(gpio);
+			/*reset chip in case of I2C mode pin is configured
+			* after power on touch chip without keep reset active
+			*/
+			gpiod_direction_output(gpio_to_desc(pdata->gpio_reset),
+						0);
+			ndelay(100);
 			gpiod_direction_output(gpio_to_desc(pdata->gpio_reset),
 						1);
 
-- 
1.9.1

